### ISSUE TITLE
TypeError: Cannot read property 'end' of undefined

### ISSUE THREAD

**Issue by Diokuz** [th0r/webpack-bundle-analyzer#4]
```
new BundleAnalyzerPlugin({ generateStatsFile: true, statsFilename: 'xxx.json' })


./node_modules/.bin/webpack-bundle-analyzer ~/online5/build/public/xxx.json -p 8888
Could't analyze webpack bundle:
TypeError: Cannot read property 'end' of undefined
```

Problem appears in production build of our application, which have no build-in server. In dev-mode all good.

**Comment by balanceiskey** [comment]
Not sure if it helps, but we get the same error when running the CLI tool against a stats.json file as well. Managed to get it running by disabling the `DedupePlugin`.

**Comment by alexindigo** [comment]
Same here. And disabling Dedupe plugin helped as well. Although it won't be the same as real bundle in my understanding.

**Comment by th0r** [comment]
@Diokuz do you use Dedupe plugin for your production build as well?

**Comment by alexindigo** [comment]
@th0r and I tried to run production build with minification, but analyzer still shows unminified sizes (in megabytes).

**Comment by kishorevarma** [comment]
I did some research on this bug, I am using code splitting in my project , so I have the chunk files  those are loaded using  **webpackJsonp** , following code is used to calculate the size of those chunks 

```
if (_.get(node, 'callee.name') === 'webpackJsonp') {
  state.sizes = getModulesSizesFromFunctionArgument(args[1]);
   return;
}

function getModulesSizesFromFunctionArgument(arg) {
  if (arg.type === 'ObjectExpression') {
    const modulesNodes = arg.properties;

    return _.transform(modulesNodes, (result, moduleNode) => {
      const moduleId = moduleNode.key.name || moduleNode.key.value;
      const moduleBody = moduleNode.value.body;

      result[moduleId] = moduleBody.end - moduleBody.start;
    }, {});
  }

  if (arg.type === 'ArrayExpression') {
    const modulesNodes = arg.elements;

    return _.transform(modulesNodes, (result, moduleNode, i) => {
      if (!moduleNode) return;

      result[i] = moduleNode.body.end - moduleNode.body.start;
    }, {});
  }

  return {};
}
```

Problem exist in these two lines, which are pre assuming all the node types are function expressions, but there can be Array Nodes or Literal Nodes for those there wont be any body in AST

```
result[moduleId] = moduleBody.end - moduleBody.start;
result[i] = moduleNode.body.end - moduleNode.body.start;
```

@th0r  above thing might be helpful to fix. this issue can be reproduced if all the asset files exist in the bundle directory.

Cheers

**Comment by th0r** [comment]
@kishorevarma thanks! Thats exactly what I was thinking of.

> but there can be Array Nodes or Literal Nodes

Do you know in which situations they can be arrays or literals?

**Comment by th0r** [comment]
@Diokuz @alexindigo @balanceiskey 
Can someone of you guys send me archive with broken bundle? Everything that webpack has created + stats file.

**Comment by Diokuz** [comment]
@th0r yes, we are using `new webpack.optimize.DedupePlugin()`. Analyzer started to work when I disabled DedupePlugin.

**Comment by PhiLhoSoft** [comment]
I had the same issue, and the culprit was indeed `webpack.optimize.DedupePlugin`.
Note that I have the error when running `webpack --profile --json` which, if I understood correctly, is unrelated to this tool, actually.
Ah, now that I updated Webpack from 1.12.13 to 1.13.3, I no longer have the issue.
Funnily, when I removed the dedupe plugin for tests, I observed the same size of files generated in production. Looks like it is no longer necessary, at least with NPM 3, I guess.

**Comment by th0r** [comment]
So, appears that if one module is duplicate of another (has the same content) `DedupePlugin` replaces it's content with just module id in output bundle.

But seems like there is another case [when module can be an object](https://github.com/webpack/webpack/blob/webpack-1/lib/optimize/DedupePlugin.js#L149).

@sokra Could you explain this case please?

**Comment by th0r** [comment]
Released `v1.4.1` with partial support for `DedupePlugin` (simple case with duplicated module ids).
Can you check it guys?

**Comment by th0r** [comment]
@Diokuz @PhiLhoSoft @alexindigo @balanceiskey 
Could you check if `v1.4.1` solves a problem?

**Comment by PhiLhoSoft** [comment]
Alas, as said, since I upgraded my libraries and tools, I can no longer reproduce the issue (which is already a good thing, no?).

**Comment by th0r** [comment]
@PhiLhoSoft I thought you had to remove `DedupePlugin` from config?

**Comment by kishorevarma** [comment]
@th0r its fixed my issue, having array expression in assets files.

**Comment by th0r** [comment]
> having array expression in assets files

Do you mean array in place of module wrapper?

**Comment by kishorevarma** [comment]
yeah exactly, following solved my issue :) thanks for that

```
function getModuleSize(node) {
  if (node.type === 'FunctionExpression') {
    return node.body.end - node.body.start;
  }

  if (node.type === 'Literal') {
    return node.end - node.start;
  }

  // Fallback for unknown node types
  return 0;
}
```

**Comment by th0r** [comment]
> But seems like there is another case when module can be an object.

@TheLarkInn Need your help here, man)

**Comment by th0r** [comment]
Released `v1.5.2` with full `DedupePlugin` support (at least hope so 😜)
